### PR TITLE
M2P-110 Adjust price mismatch feature

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
When a mismatch occurs, if the mismatch price is lower or equal than the price fault tolerance, we force the Magento order to be created with grand total is equal to the Bolt total amount. Right now, we only hard-code the price fault tolerance to be 1 cent. This PR adds the ability to allow the price fault tolerance to be configured in the Additional Configuration field. This is the same logic as M1.

E.g. Go to Store → Configuration → Sales → Payment Method → Bolt Pay → Advanced Options → Additional Configuration and configure the price fault tolerance to be like this http://prntscr.com/t0ww5a

Fixes: 
https://boltpay.atlassian.net/browse/M2P-110
https://app.asana.com/0/1177577426322770/1177852297536969

#changelog M2P-110 Adjust price mismatch feature

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Helper/Config.php [hunk 1]
```diff
@@ -1183,6 +1184,22 @@ public function shouldAdjustTaxMismatch($storeId = null)
...
+     *    "priceFaultTolerance": "1"
+     * }
+     *
+     * @param null $storeId
+     *
+     * @return string
+     */
+    public function getPriceFaultTolerance($storeId = null)
+    {
+        return (int) $this->getAdditionalConfigProperty('priceFaultTolerance', $storeId) ?: Order::MISMATCH_TOLERANCE;
```
Comment (vitaliyreznikov) [inline Helper/Config.php:1200]: We need to discuss if we want to continue use this JSON field for additional settings.
We have two other opportunities:
- add hidden setting as we did for local bolt URL
- use feature switch
cc: @matt-thomason

File: Helper/Config.php [hunk 2]
```diff
@@ -18,6 +18,7 @@
 namespace Bolt\Boltpay\Helper;
 
 use Bolt\Boltpay\Model\Api\Data\BoltConfigSettingFactory;
+use Codeception\Module\OrderHelper;
```
Comment (mccottry) [inline Helper/Config.php:21]: @johnmccombs1 Is this intentional?  I believe it is an error.

Comment (mccottry) [inline Helper/Config.php:21]: @johnmccombs1 Is this intentional?  I believe it can unintentionally cause errors in some cases.

Comment (johnmccombs1) [reply Helper/Config.php]: @mccottry it's just my typo. Updated

**Pull request merged**